server/src/main/java/org/opensearch/search/query/QueryPhase.java

```diff
@@ -165,6 +167,7 @@ public void execute(SearchContext searchContext) throws QueryPhaseExecutionExcep
             );
             searchContext.queryResult().profileResults(shardResults);
         }
+        searchContext.queryResult().setTookTimeNanos(System.nanoTime() - startTime);
```

Comment:
couple of questions:
1. The query phase time is also computed in `SearchOperationListenerExecutor` but I see that you need it in QueryResult which is stored in cache before above records the time.
2. If we store this time in `QueryResult` then while eviction it will need to deserialize each evicted value to get the tookTime and make decision of keeping it in disk tier or not. Wondering if the cache should have it as a separate wrapped object of QueryResult instead, to make it easily available ? Is there a need to output this value in the QueryResult ? If yes, then will probably use the value computed by `SearchOperationListenerExecutor` to keep it consistent with search stats.

Reply:
For 1), I wasn't aware of that - let me see if it's doable to get that stored time into the QueryResult. 
For 2), in my tiered caching policies PR that should hopefully get raised soon, there is a separate wrapper with just the important info to decide whether to add a result to a disk tier. This gets written first, before the actual result, so when it decides it can just read that wrapper and doesn't have to spend time deserializing the whole result. I will see if i can get the other value into there instead.

Reply:
For 2, not sure if I understood correctly. If you already have a wrapper concept, then that wrapper can hold to the computed `tookTime` and the computation can be done [here](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/indices/IndicesService.java#L1699) instead. That way `QueryResult` can be kept agnostic of `tookTime`